src/ContextParser.php

```
@@ -0,0 +1,299 @@
+            ));
+        }
+
+        if ($stmts === null || $stmts === []) {
+            throw new EditException(sprintf('Snippet produced no AST nodes in the "%s" context.', $context));
+        }
+
+        // A snippet can close the host construct and open its own — `} echo 1; class Y {` in
+        // the member context — and the host then holds something the extractor never expected.
+        // Check what actually came back before reaching into it, or the mismatch surfaces as
+        // an undefined-property warning and a TypeError from inside a closure.
+        $host = $spec['host'];
```

PHPStan (2.2.8)

[error] Offset 'host' does not exist on array{template: string, extract: callable(array<PhpParser\Node\Stmt>): list<PhpParser\Node>}.


(offsetAccess.notFound)